pika/adapters/base_connection.py

```diff
@@ -78,6 +78,29 @@ def __init__(self,
                                              on_open_error_callback,
                                              on_close_callback)
 
+    def __repr__(self):
+        def get_socket_repr(sock):
```

Comment:
@gmr, I am not sure whether I understand your question correctly: I considered abstracting this repr to the `Connection` class, but then realized that the stack is designed such that `Connection` doesn't know anything about sockets.

The second point regarding
> if it matches the format in the RabbitMQ management plugin.

I think you're recommending to check how the management plugin represents connections visually and match their format, right?